pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -755,33 +756,56 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 		return statusUnchanged, errors.Errorf("Could not find job %s", s.flinkController.GetLatestJobID(ctx, app))
 	}
 
-	// wait until all vertices have been scheduled and started
-	allVerticesStarted := true
-	for _, v := range job.Vertices {
-		allVerticesStarted = allVerticesStarted && (v.StartTime > 0)
-	}
-
-	if job.State == client.Running && allVerticesStarted {
-		// Update job status
-		jobStatus := s.flinkController.GetLatestJobStatus(ctx, app)
-		jobStatus.JarName = app.Spec.JarName
-		jobStatus.Parallelism = app.Spec.Parallelism
-		jobStatus.EntryClass = app.Spec.EntryClass
-		jobStatus.ProgramArgs = app.Spec.ProgramArgs
-		jobStatus.AllowNonRestoredState = app.Spec.AllowNonRestoredState
-		s.flinkController.UpdateLatestJobStatus(ctx, app, jobStatus)
-		// Update the application status with the running job info
-		app.Status.SavepointPath = ""
-		app.Status.SavepointTriggerID = ""
-		if v1beta1.IsBlueGreenDeploymentMode(app.Status.DeploymentMode) && app.Status.DeployHash != "" {
-			s.updateApplicationPhase(app, v1beta1.FlinkApplicationDualRunning)
+	// wait until all vertices have been scheduled and running
+	jobStartTimeSec := job.StartTime / 1000
```

Comment:
Is the job start time reliable ? I believe this will get reset when the jm restarts or job fails and restarts

Reply:
Hmm, good point. Let me verify it tmr.

Reply:
I have verified that the start time will not reset after job restart.

```{"jobs":[{"jid":"70ec861c2ddf20072a1d2e3f1aff3bd7","name":"nmodes","state":"RESTARTING","start-time":1680713677111,"end-time":-1,"duration":1235989,"last-modification":1680714880571,"tasks":{"total":192,"created":0,"scheduled":0,"deploying":0,"running":0,"finished":0,"canceling":0,"canceled":191,"failed":1,"reconciling":0,"initializing":0}}]}
```

The job is in RESTARTING state, but `start-time` is Wednesday, April 5, 2023 9:54:37.111 AM [GMT-07:00](https://www.epochconverter.com/timezones?q=1680713677111) DST, which is the time I triggered the deployment.